web-ui/src/main/resources/catalog/views/default/module.js

```diff
@@ -339,8 +341,36 @@
         match(/^(\/[a-zA-Z0-9]*)($|\/.*)/)[1];
       };
 
+      var availableTabs = ['general', 'contact', 'relations', 'inspire'];
+      $scope.changeTabMdView =function(newTab) {
+        if (availableTabs.indexOf(newTab) == -1) {
+          newTab = availableTabs[0];
+        }
+        $location.search('tab', newTab);
+      };
+
+      $scope.$on('tabChangeRequested', function(event, requestedTab) {
+        $scope.changeTabWithoutModifyingUrl(requestedTab);
+      });
+
+      $scope.changeTabWithoutModifyingUrl = function (newTab) {
```

Comment:
This seems too complicated for what it does. Wouldn't it be possible to just put it in the `$locationChangeSuccess` listener below?